osquery/tables/system/darwin/unified_log.mm

```diff
@@ -328,21 +354,21 @@ QueryData genUnifiedLog(QueryContext& queryContext) {
           first = false;
         }
 
+        // Escape if the rows number reached the limit
+        if (++rows_counter > max_rows)
+          break;
```

Comment:
Honestly, I care more about passing `max_rows` to the underlying search than I do in limiting it here.

Reply:
I didn't see a way to pass `max_rows` into the log enumerator (`entriesEnumeratorWithOptions`). I just moved this logic above the `if (isSequential)` block, so that we don't advance the already seen pointer for a row that's not actually processed (because it's above `max_rows`).